lib/absinthe/ecto.ex

```diff
@@ -94,11 +104,15 @@ defmodule Absinthe.Ecto do
   field :author, :user, resolve: assoc(MyApp.Repo, :author)
   ```
   """
-  def assoc(repo, association) do
-    fn parent, _, _ ->
+  def assoc(repo, association, query_fun) do
+    fn parent, args, %{context: ctx} ->
       case Map.get(parent, association) do
         %Ecto.Association.NotLoaded{} ->
-          ecto_batch(repo, parent, association)
+          if query_fun != nil do
+            ecto_batch(repo, parent, {association, fn query -> query_fun.(query, args, ctx) end})
```

Comment:
I don't understand how this is the batch key, the tuple is immediately deconstructed in the fun and there is no pattern matching at all. The batch key is computed further below as before.

Am i missing something ?